inc/item.class.php

```diff
@@ -0,0 +1,239 @@
+         echo "&nbsp;$message</a>";
+         echo "</th></tr></table>";
+      }
+   }
+
+   /**
+    * Display form
+    *
+    * @param object $item CommonDBTM instance
+    * @param array $options
+    * @return true
+    */
+   function showForm(CommonDBTM $item, $options = []) {
+      Session::checkRight('plugin_fusioninventory_printer', READ);
+
+      $fk     = getForeignKeyFieldForItemType($this->itemtype);
+      $params = [$fk => $item->getID()];
+      if (!$this->getFromDBByCrit($params)) {
+         // Add in database if not exist
+         $_SESSION['glpi_plugins_fusinvsnmp_table']
+                = getTableForItemType($this->itemtype);
+         $ID_tn = $this->add($params);
+         $this->getFromDB($ID_tn);
+      }
+
+      // Form item informations
+
+      echo "<div align='center'>";
+      echo "<form method='post' name='snmp_form' id='snmp_form'
+                 action=\"".$options['target']."\">";
+      echo "<table class='tab_cadre' cellpadding='5' width='950'>";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<th colspan='4'>";
+      echo __('SNMP information', 'fusioninventory');
+
+      echo "</th>";
+      echo "</tr>";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td align='center'>";
+      echo __('Sysdescr', 'fusioninventory');
+
+      echo "</td>";
+      echo "<td>";
+      echo "<textarea name='sysdescr' cols='45' rows='5'>";
+      echo $this->fields['sysdescr'];
+      echo "</textarea>";
+      echo "</td><td>";
+
+      echo "<table><tr>";
+      echo "<td align='center'>";
+      echo __('Last inventory', 'fusioninventory');
+      echo "</td>";
+      echo "<td>";
+      echo Html::convDateTime(
+              $this->fields['last_fusioninventory_update']);
+      echo "</td>";
+      echo "</tr>";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "</td>";
+      echo "<td align='center'>".__('SNMP authentication', 'fusioninventory')."</td>";
+      echo "<td align='center'>";
+      PluginFusioninventoryConfigSecurity::authDropdown(
+              $this->fields["plugin_fusioninventory_configsecurities_id"]);
+      echo "</td>";
+      echo "</tr>";
+      echo "</table></td></tr>";
+
+      $this->addMoreInfos($options);
+
+      echo "<tr class='tab_bg_2 center'>";
+      echo "<td colspan='4'>";
+      echo "<div align='center'>";
+      echo Html::hidden('id', ['value' => $this->fields['id']]);
+      echo Html::submit(__('Update'), ['name' => 'update']);
+      echo "</td>";
+      echo "</tr>";
+
+      echo "</table>";
+      Html::closeForm();
+      echo "</div>";
+      return true;
+   }
+
+   /**
+   * Add more info to the tab if needed
+   * @since 9.2+2.0
+   *
+   * @param array $options display options
+   */
+   function addMoreInfos($options = []) {
+
+   }
+
+   /**
+   * Get a FI class instance representing an itemtype
+   * @since 9.2+2.0
+   *
+   * @param array $options display options
+   */
+   static function getFIItemClassInstance($itemtype) {
```

Comment:
Maybe should we add a message in logs when it is called with an unknown item type?